tests/test_builtin_casters.py

```diff
@@ -286,14 +300,20 @@ def cant_convert(v):
     assert noconvert(7) == 7
     cant_convert(3.14159)
     assert convert(DeepThought()) == 42
-    require_implicit(DeepThought())
+    requires_conversion(DeepThought())
+    assert convert(DoubleThought()) == 0  # Fishy; `int(DoubleThought)` == 42
```

Comment:
This is not great, but kind of a consequence of us saying "everything with `__index__` is already an `int`, so don't try converting".